package/Shaders/Lighting.hlsl

```diff
@@ -1830,10 +1835,19 @@ PS_OUTPUT main(PS_INPUT input, bool frontFace
 	float dirDetailShadow = 1.0;
 	float dirShadow = 1.0;
 	float parallaxShadow = 1;
-#	if defined(SOFT_LIGHTING) || defined(BACK_LIGHTING) || defined(RIM_LIGHTING)
+#	if defined(DEFERRED)
+
+#		if defined(SOFT_LIGHTING) || defined(BACK_LIGHTING) || defined(RIM_LIGHTING)
 	if (!inDirShadow && dirLightAngle > 0.0) {
+#		else
+	if (!inDirShadow && PixelShaderDescriptor) {
+#		endif
 #	else
-	if (!inDirShadow) {
+#		if defined(SOFT_LIGHTING) || defined(BACK_LIGHTING) || defined(RIM_LIGHTING)
+	if (!inDirShadow && dirLightAngle > 0.0 && (PixelShaderDescriptor & _InWorld || !FrameParams.y)) {
+#		else
+	if (!inDirShadow && PixelShaderDescriptor & (PixelShaderDescriptor & _InWorld || !FrameParams.y)) {
```

Comment:
Better make the condition a bool for clarity, since the defines seems separatable

Reply:
also, why is the last condition a bitwise &?

Reply:
```cpp
	bool theCondition = !inDirShadow;
#	if defined(SOFT_LIGHTING) || defined(BACK_LIGHTING) || defined(RIM_LIGHTING)
	theCondition = theCondition && (dirLightAngle > 0.0);
#	else
	theCondition = theCondition && PixelShaderDescriptor;
#	endif
#	if !defined(DEFERRED)
	theCondition = theCondition && (PixelShaderDescriptor & _InWorld || !FrameParams.y);
#	endif
	if(theCondition)
```